src/Peachpie.CodeAnalysis/FlowAnalysis/ExpressionAnalysis.cs

```diff
@@ -1551,7 +1584,15 @@ public override T VisitStaticFunctionCall(BoundStaticFunctionCall x)
 
                 candidates = Construct(candidates, x);
 
-                x.TargetMethod = new OverloadsList(candidates).Resolve(this.TypeCtx, x.ArgumentsInSourceOrder, VisibilityScope);
+                var method = new OverloadsList(candidates).Resolve(this.TypeCtx, x.ArgumentsInSourceOrder, VisibilityScope);
+                if ((method is MissingMethodSymbol || method is InaccessibleMethodSymbol)
+                    && type.LookupMember<IMethodSymbol>(Name.SpecialMethodNames.CallStatic.Value) != null)
+                {
+                    // __callStatic at runtime solves both inaccessible and missing method problems
+                    method = null;
```

Comment:
nice!
please add note like ~ `// TODO: remember and emit call to __callstatic directly (CallStaticMethodSymbol?)`

Reply:
OK, added in 2eee6f3ae66e8a265d3704329c9789ab05bc34bd.